source/server/listener_manager_impl.cc

```diff
@@ -178,7 +178,12 @@ ProdListenerComponentFactory::createUdpListenerFilterFactoryListImpl(
     ENVOY_LOG(debug, "  config: {}",
               MessageUtil::getJsonStringFromMessageOrError(
                   static_cast<const Protobuf::Message&>(proto_config.typed_config())));
-
+    if (proto_config.config_type_case() ==
+        envoy::config::listener::v3::ListenerFilter::ConfigTypeCase::kConfigDiscovery) {
+      RELEASE_ASSERT(false, "For UDP listener, the dynamic listener filter configuration "
```

Comment:
Not sure we should use RELEASE_ASSERT, I would prefer raise a exception here.

Reply:
ACK